File: 1._$A._HP_ITEL_P80_LATIHAN_CODING/Latihan_ayo_tes.py
print ("\n Kalkulator dasar \n")

def tambah (x, y):
    return x + y


def kurang (x, y):
    return x - y


def kali (x, y):
    return x * y


def bagi (x, y):
    return x / y


def pangkat (x, y):
    return x ** y


hasil_a = tambah (10, 10)
hasil_b = kurang (10, 5)
hasil_c = kali (10, 10)
hasil_d = bagi (10, 5)
hasil_e = pangkat (10, 10)


hitung = f"""
- Tambah  = {hasil_a}
- Kurang  = {hasil_b}
- Kali    = {hasil_c}
- Bagi    = {hasil_d}
- Pangkat = {hasil_e}
"""

print (hitung)


print ("\n --- batas --- \n")




print ("\n Tipe Data Pemrograman \n")

teks = "Hujan"
angka = 12
desimal = 2.12
cek = True
char = 'a'
kosong = None


tipe = f"""
- Teks    : {teks}
- Angka   : {angka}
- Desimal : {desimal}
- Cek     : {cek}
- Char    : {char}
- Kosong  : {kosong}
"""


print (tipe)


print ("\n --- batas --- \n")




print ("\n Fungsi dengan percabangan dasar \n")

def run (a):
    
    if a >= 5:
        print (f"Besar, angka a = {a}")
        
    else:
        print (f"Kecil, angka a = {a}")
        
run (10)
run (9)
run (8)
run (7)
run (6)
run (5)
run (4)
run (3)
run (2)
run (1)


print ("\n --- batas -- \n")




print ("\n Fungsi dengan Percabangan Lanjutan \n")

def run (e):
     
     if e >= 8:
          print (f"Besar, angka e = {e}")
          
     elif e >= 5:
          print (f"Tengah, angka e = {e}")
          
     else:
          print (f"Kecil, angka e = {e}")
          
run (10)
run (9)
run (8)
run (7)
run (6)
run (5)
run (4)
run (3)
run (2)
run (1)


print ("\n --- batas --- \n")




print ("\n Fungsi dengan Percabangan Skor \n")

def skor (f):
     
     if f >= 95:
          print (f"A, skor = {f}")
          
     elif f >= 90:
          print (f"B, skor = {f}")
          
     elif f >= 80:
          print (f"C, skor = {f}")
          
     elif f >= 70:
          print (f"D, skor = {f}")
     
     elif f >= 60:
          print (f"E, skor = {f}")
          
     elif f >= 50:
          print (f"F, skor = {f}")
          
     else:
          print (f"Jelek amat, skor = {f}")
          
skor (100)
skor (90)
skor (80)
skor (70)
skor (60)
skor (50)
skor (40)


print ("\n --- batas --- \n")




print ("\n Fungsi dengan Percabangan Nested \n")

def run (w):
     
     cek = True
     
     if w >= 5:
          if cek:
               print (f"Besar, angka w = {w}")
               
     else:
          print (f"Kecil, angka w = {w}")
          
          
run (10)
run (8)
run (7)
run (5)
run (4)
run (3)
run (2)


print ("\n --- batas --- \n")




print ("\n Fungsi dengan Switch Case dengan Int \n")

def rust (l):
     
     match (l):
          
          case 1:
               print ("Angka 1")
               
          case 2:
               print ("Angka 2")
               
          case 3:
               print ("Angka 3")
               
          case 4:
               print ("Angka 4")
               
          case 5:
               print ("Angka 5")
               
          case _:
               print ("Angka lain")
               
rust (1)
rust (2)
rust (3)
rust (4)
rust (5)
rust (6)


print ("\n --- batas --- \n")




print ("\n Fungsi dengan Swich Case dengan String \n")

def warna (e):
     
     match (e):
          
          case "Merah":
               print ("Warna Merah")
               
          case "Kuning":
               print ("Warna Kuning")
               
          case "Hijau":
               print ("Warna Hijau")
               
          case _:
               print ("Warna lain")
               
warna ("Merah")
warna ("Kuning")
warna ("Hijau")
warna ("Hitam")


print ("\n --- batas --- \n")




print ("\n Fungsi + Percabangan Dasar + Mencari angka terkecil \n")

def kecil (c, d):
     
     if c <= d:
          return c 
     
     else:
          return d
     
hasil_1 = kecil (10, 5)
hasil_2 = kecil (5, 10)
hasil_3 = kecil (23, 5)
hasil_4 = kecil (7, 10)


print (hasil_1)
print (hasil_2)
print (hasil_3)
print (hasil_4)


print ("\n --- batas --- \n")




print ("\n Fungsi + Percabangan Dasar + Mencari angka terbesar \n")

def besar (l, m):
     
     if l >= m:
          return l
     
     else:
          return m
     
hasil_der = besar (20, 5)
hasil_ver = besar (90, 3)
hasil_wer = besar (34, 9)
hasil_rer = besar (2, 12)


print (hasil_der)
print (hasil_ver)
print (hasil_wer)
print (hasil_rer)


print ("\n --- batas --- \n")




print ("\n Fungsi + Percabangan Dasar + Mencari Angka Besar \n")

for a in range (15):
     if a == 10:
          continue
     print (a)
     
     
print ("\n --- batas --- \n")




print ("\n For + If lanjutan 2 \n")

for i in range (15):
     if i == 10:
          break
     print (i)
     
     
print ("\n --- batas --- \n")




print ("\n Array + For + If Lanjutan \n")

daf = ["Rans", "Fans", "Fons", "Kesh", "Best"]

for i in daf:
     if i == "Fons":
          continue
     print (i)
     
     
print ("\n --- batas --- \n")




print ("\n Array + For + If Lanjutan \n")

der = ["Plan", "Run", "Compiler", "Rust", "Int", "Float"]

for k in der:
     if k == "Compiler":
          break
     print (i)
     
     
print ("\n --- batas --- \n")




print ("\n Struktur Data \n")

data = {
     "teks" : "halo dunia",
     "angka" : 12,
     "desimal" : 3.14,
     "cek" : True,
     "kosong" : None,
}

print ("Teks :", data ["teks"])

print ("Angka :", data ["angka"])

print ("Desimal :", data ["desimal"])

print ("Cek :", data ["cek"])

print ("Kosong :", data ["kosong"])

File: 1._$A._HP_ITEL_P80_LATIHAN_CODING/Latihan_ayo_tes.js
console.log ("\n Kalkulator dasar \n")

function tambah (x, y) {
    return x + y
}


function kurang (x, y) {
    return x - y
}


function kali (x, y) {
    return x * y
}


function bagi (x, y) {
    return x / y
}


function pangkat (x, y) {
    return x ** y
}


hasil_a = tambah (10, 10)
hasil_b = kurang (10, 5)
hasil_c = kali (10, 10)
hasil_d = bagi (10, 5)
hasil_e = pangkat (10, 10)


hitung = `
- Tambah  = ${hasil_a}
- Kurang  = ${hasil_b}
- Kali    = ${hasil_c}
- Bagi    = ${hasil_d}
- Pangkat = ${hasil_e}
`

console.log (hitung)


console.log ("\n --- batas --- \n")




console.log ("\n Tipe Data Pemrograman \n")

var teks = "Hujan"
var angka = 12
var desimal = 2.12
var cek = true
var char = 'a'
var kosong = null


tipe = `
- Teks    : ${teks}
- Angka   : ${angka}
- Desimal : ${desimal}
- Cek     : ${cek}
- Char    : ${char}
- Kosong  : ${kosong}
`


console.log (tipe)


console.log ("\n --- batas --- \n")




console.log ("\n Fungsi dengan Percabangan Dasar \n")

function run (a) {
    
    if (a >= 5) {
        console.log (`Besar, angka a = ${a}`)
    }
    
    else {
        console.log (`Kecil, angka a = ${a}`)
    }
}

run (10)
run (9)
run (8)
run (7)
run (6)
run (5)
run (4)
run (3)
run (2)
run (1)


console.log ("\n --- batas -- \n")




console.log ("\n Fungsi dengan percabangan lanjutan \n")

function der (t) {
    
    if (t >= 8) {
        console.log (`Besar, angka t = ${t}`)
    }
    
    else if (t >= 5) {
        console.log (`Tengah, angka t = ${t}`)
    }
    
    else {
        console.log (`Kecil, angka t = ${t}`)
    }
}

der (10)
der (9)
der (8)
der (7)
der (6)
der (5)
der (4)
der (3)
der (2)
der (1)


console.log ("\n --- batas --- \n")



console.log ("\n Fungsi dengan Percabangan Skor \n")

function skor (f) {

     if (f >= 95) {
          console.log (`A, skor = ${f}`)
     }

     else if (f >= 90) {
          console.log (`B, skor = ${f}`)
     }

     else if (f >= 80) {
          console.log (`C, skor = ${f}`)
     }

     else if (f >= 70) {
          console.log (`D, skor = ${f}`)
     }

     else if (f >= 60) {
          console.log (`E, skor = ${f}`)
     }

     else if (f >= 50) {
          console.log (`F, skor = ${f}`)
     }

     else {
          console.log (`Jelek amat, skor = ${f}`)
     }
}

skor (100)
skor (90)
skor (80)
skor (70)
skor (60)
skor (50)
skor (40)


console.log ("\n --- batas --- \n")





console.log ("\n Fungsi dengan Percabangan Nested 1 \n")

function far (k) {

     cek = true

     if (k >= 5) {
          if (cek) {
               console.log (`Besar, angka k = ${k}`)
          }
     }

     else {
          console.log (`Kecil, angka k = ${k}`)
     }
}

far (10)
far (9)
far (8)
far (7)
far (6)
far (5)
far (4)
far (3)
far (2)
far (1)

console.log ("\n --- batas --- \n")




console.log ("\n Switch Case  1 dengan int \n")

function kuo (p) {
    
    switch (p) {
        
        case 1:
            console.log ("Angka 1")
            break
            
        case 2:
            console.log ("Angka 2")
            break
            
        case 3:
            console.log ("Angka 3")
            break
            
        case 4:
            console.log ("Angka 4")
            break
            
        case 5:
            console.log ("Angka 5")
            break
            
        default:
        console.log ("Angka lain")
    }
}

kuo (1)
kuo (2)
kuo (3)
kuo (4)
kuo (5)
kuo (6)


console.log ("\n --- batas --- \n")




console.log ("\n Switch Case 2 dengan String \n")

function warna (b) {
    
    switch (b) {
        
        case "Merah":
            console.log ("Warna merah")
            break
            
        case "Kuning":
            console.log ("Warna kuning")
            break
            
        case "Hijau":
            console.log ("Warna hijau")
            break
            
        default:
        console.log ("Warna lain")
    }
}

warna ("Merah")
warna ("Kuning")
warna ("Hijau")
warna ("Hitam")


console.log ("\n --- batas --- \n")




console.log ("Fungsi return + percabangan dasar + mencari angka terkecil \n")

function angkan (i, p) {
    
    if (i < p) {
        return i
    }
    
    else {
        return p
    }
}

hasil_1 = angkan (1, 91)
hasil_2 = angkan (2, 31)
hasil_3 = angkan (3, 21)
hasil_4 = angkan (7, 11)
hasil_5 = angkan (2, 19)

console.log (hasil_1)
console.log (hasil_2)
console.log (hasil_3)
console.log (hasil_4)
console.log (hasil_5)


console.log ("\n --- batas --- \n")




console.log ("\n Fungsi return + percabangan dasar + mencari angka terbesar \n")

function desk (a, b) {
    
    if (a > b) {
        return a
    }
    
    else {
        return b
    }
}

hasil_k = desk (1, 10)
hasil_j = desk (3, 34)
hasil_g = desk (2, 24)
hasil_v = desk (7, 79)
hasil_x = desk (6, 27)

console.log (hasil_k)
console.log (hasil_j)
console.log (hasil_g)
console.log (hasil_v)
console.log (hasil_x)


console.log ("\n --- batas --- \n")




console.log ("\n For If Lanjutan \n")

for (a = 0; a < 11; a++) {
     if (a == 5) {
          continue
     }

     console.log (a)
}


console.log ("\n --- batas --- \n")



console.log ("\n For If lanjutan 2 \n")

for (r = 0; r < 11; r++) {
     if (r == 5) {
          break
     }

     console.log (r)
}

console.log ("\n --- batas --- \n")




console.log ("\n Array + For + If Lanjutan \n")

daf = ["Ranur", "Frank", "Ron", "Ajax", "Ajam"]

for (var a of daf) {
     if (a == "Ron") {
          continue
     }

     console.log (a)
}


console.log ("\n --- batas --- \n")



console.log ("\n Array + For + IF Lanjutan 1 \n")

ref = ["Hans", "Dans", "Rans", "Rons", "Fons"]

for (var b of ref) {
     if (b == "Rans") {
          break
     }

     console.log (b)
}


console.log ("\n --- batas --- \n")




console.log ("\n Struktur Data \n")

var data = {
     "teks" : "halo dunia",
     "angka" : 12,
     "desimal" : 3.14,
     "cek" : True,
     "kosong" : None,
}

console.log ("Teks :", data ["teks"])

console.log ("Angka :", data ["angka"])

console.log ("Desimal :", data ["desimal"])

console.log ("Cek :", data ["cek"])

console.log ("Kosong :", data ["kosong"])